### PR TITLE
EEMS unit tests added

### PR DESCRIPTION
**Pull request by lusassl-msft**
**Description:**
Added Pester unit tests for Exchange Emergency Mitigation Service (EEMS).

### CODE REVIEW

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        )
+    }
+}
+
+Describe "Testing Get-ExchangeEmergencyMitigationServiceState.ps1" {
+
+    Context "Exchange Emergency Mitigation Service Default State" {
+
+        BeforeAll {
+            Mock Invoke-WebRequest -MockWith { return Import-Clixml $Script:parentPath\Tests\InvokeWebRequestEEMS.xml }
```
Comment (dpaulson45) [inline Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1:32]: Move the `BeforeAll` action under `Context` to under `Describe` this avoids you having to run the same `Mock` instructions multiple times. Only set a different `Mock` under the `Context` if it is different from the default testing. 

Place the most common `Mock` instruction in the `Describe` and only override it in the `Context` when you need it.

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            Mock Get-Service -MockWith { return Import-Clixml $Script:parentPath\Tests\GetServiceMSExchangeMitigationRunning.xml } `
+                -ParameterFilter { $ComputerName -eq $Script:Server }
+
+            $customObject = ([PSCustomObject]@{
+                    ComputerName       = $Script:Server
```
Comment (dpaulson45) [inline Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1:37]: Recommend changing this to a function that you can call and pass the parameters that are changing in your `Context`

File: Diagnostics/HealthChecker/Tests/HealthChecker.Tests.ps1 [hunk 3]
```diff
@@ -338,6 +338,14 @@ Describe "Testing Health Checker by Mock Data Imports" {
             SetActiveDisplayGrouping "Security Settings"
             TestObjectMatch "AMSI Enabled" "False" -WriteType "Yellow"
         }
+
+        It "EEMS Enabled And OCS Reachable" {
```
Comment (dpaulson45) [inline Diagnostics/HealthChecker/Tests/HealthChecker.Tests.ps1:342]: Was the `Mock` added to import the XML data for this already?

**Review by dpaulson45 — CHANGES_REQUESTED** [review]

File: Diagnostics/HealthChecker/Tests/HealthChecker.Tests.ps1 (hunk 3 above)
Comment (lusassl-msft) [reply Diagnostics/HealthChecker/Tests/HealthChecker.Tests.ps1:342]: Yes: `Mock Get-Service { return Import-Clixml "$Script:MockDataCollectionRoot\Exchange\GetServiceMitigation.xml" }`.
Telemetry data is in organization configuration which is also available. Pattern service check is performed in context of the machine which runs the unit tests so outbound internet connectivity is possible (we receive a 200 OK).

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1 (hunk 1 above)
Comment (lusassl-msft) [reply Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1:32]: Good point. Will do that.

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1 (hunk 2 above)
Comment (lusassl-msft) [reply Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Tests/Get-ExchangeEmergencyMitigationServiceState.Tests.ps1:37]: I've just changed the properties within the `BeforeAll` under `Context`. Looks like it do the job well.

**Review by dpaulson45 — APPROVED** [review]

**Pull request merged**